toolkits/google/arcade_google/tools/gmail.py

```diff
@@ -329,4 +336,109 @@ async def list_emails(
         except Exception as e:
             print(f"Error reading email {msg['id']}: {e}")
 
-    return json.dumps({"emails": emails})
+    return {"emails": emails}
+
+
+@tool(
+    requires_auth=Google(
+        scopes=["https://www.googleapis.com/auth/gmail.readonly"],
+    )
+)
+async def search_threads(
+    context: ToolContext,
+    page_token: Annotated[
+        Optional[str], "Page token to retrieve a specific page of results in the list"
```

Comment:
Q: Do we want to expose `page_token` to the LLM? It looks like it's used internally by our function to paginate up to `max_results`. Thinking out loud, I guess if the LLM specified a page token, it would start at that page and then continue up to `max_results`?